config/config.go

```diff
@@ -41,15 +67,44 @@ func (c Config) Write() error {
 	logrus.Infof("Saving config to %s", c.Path)
 	p := c.Path
 	c.Path = ""
-	output, err := os.Create(p)
+	output, err := os.OpenFile(p, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
 	if err != nil {
 		return err
 	}
 	defer output.Close()
 
+	if err := warnInsecurePermissions(p); err != nil {
+		return err
+	}
+
 	return json.NewEncoder(output).Encode(c)
 }
 
+var warned bool
+
+// The configuration file may have sensitive information (eg: creds) so
+// warn against possibly insecure permissions.
+func warnInsecurePermissions(path string) error {
```

Comment:
I'd avoid this global state with the `warned` variable and just return a warning as a formatted string. Then you can log the warning if needed. It's easier to test and use this way.

Reply:
The `warned` global variable has been removed as suggested.